src/unistd.rs

```diff
@@ -360,6 +360,25 @@ pub fn pipe() -> Result<(RawFd, RawFd)> {
     }
 }
 
+// libc only defines `pipe2` in `libc::notbsd`.
+#[cfg(any(target_os = "linux",
+          target_os = "android",
+          target_os = "emscripten"))]
+pub fn pipe2(flags: OFlag) -> Result<(RawFd, RawFd)> {
+    unsafe {
```

Comment:
Could you use `unsafe` as finegrained as possible, even if you need more instances.